include/fastdds/rtps/RTPSDomain.h

```diff
@@ -134,6 +135,27 @@ class RTPSDomain
             WriterHistory* hist,
             WriterListener* listen = nullptr);
 
+    /**
+     * Create a RTPSWriter in a participant using a custom payload pool.
+     * @param p Pointer to the RTPSParticipant.
+     * @param watt Writer Attributes.
+     * @param payload_pool Shared pointer to the IPayloadPool
+     * @param change_pool Shared pointer to the IChangePool
+     * @param hist Pointer to the WriterHistory.
+     * @param listen Pointer to the WriterListener.
+     * @return Pointer to the created RTPSWriter.
+     *
+     * \warning The returned pointer is invalidated after a call to removeRTPSWriter() or stopAll(),
+     *          so its use may result in undefined behaviour.
+     */
+    RTPS_DllAPI static RTPSWriter* createRTPSWriter(
```

Comment:
Could we also have an overload of this receiving a `const EntityId_t&`, it will just forward the call to `RTPSDomainImpl::create_rtps_writer`